lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -0,0 +1,92 @@
+"""
+Tests for Discussion API views
+"""
+from datetime import datetime
+import json
+
+import mock
+from pytz import UTC
+
+from django.core.urlresolvers import reverse
+
+from student.tests.factories import CourseEnrollmentFactory, UserFactory
+from util.testing import UrlResetMixin
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class CourseTopicsViewTest(UrlResetMixin, ModuleStoreTestCase):
+    """Tests for CourseTopicsView"""
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUp(self):
+        super(CourseTopicsViewTest, self).setUp()
+        self.maxDiff = None  # pylint: disable=invalid-name
+        self.course = CourseFactory.create(
+            org="x",
+            course="y",
+            run="z",
+            start=datetime.now(UTC),
+            discussion_topics={"Test Topic": {"id": "test_topic"}}
+        )
+        self.password = "password"
+        self.user = UserFactory.create(password=self.password)
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+        self.url = reverse("course_topics", kwargs={"course_id": unicode(self.course.id)})
+        self.client.login(username=self.user.username, password=self.password)
+
+    def assert_response_correct(self, response, expected_status, expected_content):
+        """
+        Assert that the response has the given status code and parsed content
+        """
+        self.assertEqual(response.status_code, expected_status)
+        parsed_content = json.loads(response.content)
+        self.assertEqual(parsed_content, expected_content)
+
+    def test_not_authenticated(self):
```

Comment:
Similar suggestion: consider creating a test mixin for testing `test_not_enrolled`, `test_non_existent_course`, and `test_not_authenticated`, as they will be needed by the other Forums APIs.  This can be done in a subsequent PR.